### PR TITLE
feat: make default limit configurable

### PR DESCRIPTION
**Pull request by magnew**
Closes: #5082 

### Description:

Adds an environment variable to make the default limit configurable. I believe this is the first explore default param that is set in the env config, so there is a little plumbing to make that work. 

Note that this is *not* currently plumbed into the SQL runner or Semantic viewer, but that could be added here or in a follow up. 

To test: 
- set `LIGHTDASH_QUERY_DEFAULT_LIMIT` 
- check that the new default is used in explores

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/providers/ExplorerProvider.tsx [hunk 1]
```diff
@@ -1325,12 +1326,27 @@ export const ExplorerProvider: FC<
...
+            unsavedChartVersion: {
+                ...defaultState.unsavedChartVersion,
+                metricQuery: {
+                    ...defaultState.unsavedChartVersion.metricQuery,
+                    ...(defaultLimit !== undefined && { limit: defaultLimit }),
+                },
+            },
+        }),
+        [defaultLimit],
+    );
```
Comment (magnew) [inline packages/frontend/src/providers/ExplorerProvider.tsx:1345]: Populating this provider from an env config for the first time

**Comment by hamzahc1** [comment]
@magnew - the original ticket proposed having this in the project settings, is there a reason this is an env variable instead?

**Comment by rephus** [comment]
Updated render environment with LIGHTDASH_QUERY_DEFAULT_LIMIT=5000

**Comment by rephus** [comment]
> @magnew - the original ticket proposed having this in the project settings, is there a reason this is an env variable instead?

This limit is there to ensure people don't run large queries by default, plus all our limits are ENV variables, so I think it makes sense to keep this as an ENV variable too. The use case we're trying to cover doesn't require this to be configurable by the admin.

**Comment by magnew** [comment]
> > @magnew - the original ticket proposed having this in the project settings, is there a reason this is an env variable instead?
> 
> This limit is there to ensure people don't run large queries by default, so it makes sense to keep this internal; plus all our limits are ENV variables, so I think it makes sense to keep this as an ENV variable too. The use case we're trying to cover doesn't require this to be configurable by the admin.

@hamzahc1 Yea, Javi suggested doing it this way to cover the Tinder use case. We pulled in this ticket, which is kind of old and didn't modify it. I think what we've done here will work for Tinder, but we could consider adding a UI setting on top.

File: packages/frontend/src/pages/MinimalSavedExplorer.tsx [hunk 2]
```diff
@@ -120,6 +122,7 @@ const MinimalSavedExplorer: FC = () => {
                       }
                     : undefined
             }
+            defaultLimit={health.data?.query.defaultLimit}
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/pages/MinimalSavedExplorer.tsx]: I put it here just for completeness in initialising the provider, but you're right it's not needed when there is not a run button. It works fine without it so I have removed it.

I don't think it's an extra call though, the MinimalExplore already makes this call in this file.

**Review by hamzahc1 — APPROVED** [review]
Thanks for explaining Marshall - looks good!

**Pull request merged**